components/avatar/avatar.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+  <span :class="classString">
+    <img v-if="src && isImgExist" :src="src" @error="handleImgLoadError" />
+    <Icon v-else-if="icon" :type="icon" />
+    <span v-else ref="children" :class="`${ prefixCls }-string`" :style="childrenStyle">
+      <slot></slot>
+    </span>
+  </span>
+</template>
+
+<script>
+import Icon from '../icon'
+import { oneOf } from '../_util/proptype'
+
+const SIZEMAP = {
+  'small': 'sm',
+  'large': 'lg'
+}
+
+export default {
+  name: 'avatar',
+  props: {
+    icon: String,
+    prefixCls: {
+      type: String,
+      default: 'ant-avatar'
+    },
+    shape: {
+      validator: function (value) {
+        return oneOf(value, ['circle', 'square'])
+      },
+      default: 'circle'
+    },
+    size: {
+      validator: function (value) {
+        return oneOf(value, ['large', 'small', 'default'])
+      },
+      default: 'default'
+    },
+    src: {
+      type: String
+    }
+  },
+  data () {
+    return {
+      scale: 1,
+      isImgExist: true,
+      offsetWidth: 0
+    }
+  },
+  computed: {
+    childrenStyle () {
+      return {
+        transform: `scale(${this.scale})`,
+        position: 'absolute',
+        display: 'inline-block',
+        left: `calc(50% - ${Math.round(this.offsetWidth / 2)}px)`
+      }
+    },
+    classString () {
```

Comment:
suggest rename it to classes to keep unify with other component

Reply:
ok